src/test/java/com/datastax/oss/protocol/internal/RequestFrameCodecTest.java

```diff
@@ -53,15 +54,20 @@ public void should_encode_request_frame(
       boolean tracing,
       Map<String, ByteBuffer> customPayload) {
 
+    List<String> eventTypes =
+        NullAllowingImmutableList.of(
+            ProtocolConstants.EventType.SCHEMA_CHANGE, ProtocolConstants.EventType.STATUS_CHANGE);
+    Register registerRequest = new Register(eventTypes);
+
     FrameCodec<MockBinaryString> frameCodec =
         new FrameCodec<>(
             primitiveCodec,
             compressor,
             (FrameCodec.CodecGroup)
-                registry -> registry.addEncoder(new MockOptionsCodec(protocolVersion)));
+                registry -> registry.addEncoder(new MockRegisterCodec(protocolVersion)));
 
     Frame frame =
-        Frame.forRequest(protocolVersion, STREAM_ID, tracing, customPayload, Options.INSTANCE);
+        Frame.forRequest(protocolVersion, STREAM_ID, tracing, customPayload, registerRequest);
```

Comment:
The REGISTER request looked like the next simplest request frame that was eligible to be compressed for the purposes of this test.